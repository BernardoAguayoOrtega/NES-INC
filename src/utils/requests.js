// import axios
import axios from 'axios';

// create the instance
export const newsAPI = axios.create({
  baseURL: 'https://backend-platzi-news.herokuapp.com/api',
});

// create the instance
export const searchApi = axios.create({
  baseURL: 'https://backend-platzi-news.herokuapp.com/search',
});

/**
 * @description return the route for make the call, also you can add other variable
 * @param {string} route
 */
export const callNewsApi = (route = 'news') => {
  return (variable = '') => `/${route}${variable}`;
};
