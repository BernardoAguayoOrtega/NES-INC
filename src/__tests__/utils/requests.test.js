// import app component
import { newsAPI, callNewsApi } from '../../utils/requests.js';

describe('request.js', () => {

  it('is match snapshot newsAPI?', () => {
    expect(newsAPI)
      .toMatchSnapshot()
  });

  it('is match snapshot callNewsApi?', () => {
    expect(callNewsApi)
      .toMatchSnapshot()
  });

  it('is match snapshot callNewsApi?', () => {
    const output = callNewsApi('random')
    expect(output).toMatchSnapshot()
  });

});
