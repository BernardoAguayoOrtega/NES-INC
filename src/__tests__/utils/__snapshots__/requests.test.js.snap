// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request.js is match snapshot callNewsApi? 1`] = `[Function]`;

exports[`request.js is match snapshot callNewsApi? 2`] = `[Function]`;

exports[`request.js is match snapshot newsAPI? 1`] = `[Function]`;
